File: q3/generate_train.m
%x=importdata('20ng-rec_talk.txt');
%x = x(randperm(size(x,1)),:);

%f=fopen('shuffled_features1.txt','w');
%for i=1:length(x)
%fprintf(f, '%s\n',x{i} );
%end

f = fopen('features.txt');

f1 = fopen('train1.txt','w')
f2 = fopen('test1.txt','w')
i=0;
while ~feof(f)
	i=i+1
	line = fgetl(f);

 if(1<=i<=1446)
 	fprintf(f2, '%s\n', line);
 else
fprintf(f1, '%s\n', line);
end
end

fclose(f);
fclose(f1);
fclose(f2);

f = fopen('features.txt');

f1 = fopen('train2.txt','w')
f2 = fopen('test2.txt','w')
i=0;
while ~feof(f)
	i=i+1
	line = fgetl(f);

 if(1447<=i<=2892)
 	fprintf(f2, '%s\n', line);
 else
fprintf(f1, '%s\n', line);
end
end

fclose(f);
fclose(f1);
fclose(f2);

f = fopen('features.txt');

f1 = fopen('train3.txt','w')
f2 = fopen('test3.txt','w')
i=0;
while ~feof(f)
	i=i+1
	line = fgetl(f);

 if(2893<=i<=4338)
 	fprintf(f2, '%s\n', line);
 else
fprintf(f1, '%s\n', line);
end
end

fclose(f);
fclose(f1);
fclose(f2);

f = fopen('features.txt');

f1 = fopen('train4.txt','w')
f2 = fopen('test4.txt','w')
i=0;
while ~feof(f)
	i=i+1
	line = fgetl(f);

 if(4339<=i<=5784)
 	fprintf(f2, '%s\n', line);
 else
fprintf(f1, '%s\n', line);
end
end

fclose(f);
fclose(f1);
fclose(f2);

f = fopen('features.txt');

f1 = fopen('train5.txt','w')
f2 = fopen('test5.txt','w')
i=0;
while ~feof(f)
	i=i+1
	line = fgetl(f);

 if(5785<=i<=7230)
 	fprintf(f2, '%s\n', line);
 else
fprintf(f1, '%s\n', line);
end
end

fclose(f);
fclose(f1);
fclose(f2);

File: q3/generate_train.py
import random
with open('features.txt','r') as source:
    data = [ (random.random(), line) for line in source ]
data.sort()
with open('shuffled_features.txt','w') as target:
    for _, line in data:
        target.write( line )


f = open('shuffled_features.txt','r')
f1 = open('train1.txt','w')
f2=open('test1.txt','w')
i=0
for line in f:
 i=i+1
 if(1<=i<=1446):
 	f2.write(line)
 else:
 	f1.write(line)

f1.close()
f2.close()
f.close()


f = open('shuffled_features.txt','r')
f1 = open('train2.txt','w')
f2=open('test2.txt','w')
i=0
for line in f:
 i=i+1
 if(1447<=i<=2892):
 	f2.write(line)
 else:
 	f1.write(line)

f1.close()
f2.close()
f.close()

f = open('shuffled_features.txt','r')
f1 = open('train3.txt','w')
f2=open('test3.txt','w')
i=0
for line in f:
 i=i+1
 if(2893<=i<=4338):
 	f2.write(line)
 else:
 	f1.write(line)

f1.close()
f2.close()
f.close()

f = open('shuffled_features.txt','r')
f1 = open('train4.txt','w')
f2=open('test4.txt','w')
i=0
for line in f:
 i=i+1
 if(4339<=i<=5784):
 	f2.write(line)
 else:
 	f1.write(line)

f1.close()
f2.close()
f.close()

f = open('shuffled_features.txt','r')
f1 = open('train5.txt','w')
f2=open('test5.txt','w')
i=0
for line in f:
 i=i+1
 if(5785<=i<=7230):
 	f2.write(line)
 else:
 	f1.write(line)

f1.close()
f2.close()
f.close()